client/src/components/DeleteReview.js

```diff
@@ -0,0 +1,39 @@
+import React, { useState } from "react";
+
+const DeleteButton = (props) =>{
+  const { reviewId, reviews, restaurant, setRestaurant } = props
+  
+  const deleteReview = async () => {
+    try {
+        const response = await fetch (`/api/v1/reviews/${reviewId}`, {
+        method: "delete",
+        headers: new Headers({
+          "Content-Type": "application/json"
+        })
+      })
+       if(!response.ok) {
+        const errorMessage = `${response.status} (${response.statusText})`
+        const error = new Error(errorMessage)
+        throw(error)
+      }
+      const returnedReviews = reviews.filter(review => review.id !== reviewId)
+      setRestaurant({
+        ...restaurant,
+        reviews: returnedReviews
+      })
+      return { message: "reeview has been deleted" }
```

Comment:
typo in `review` - we would expect this message to be part of the `response` that is returned from the router, not something that is created for the return of this function

it also looks like this return is not used or needed since above you are setting state